gk/main.c

```diff
@@ -659,6 +657,50 @@ add_ggu_policy(struct ggu_policy *policy,
 	}
 }
 
+/*
+ * Flushes flow entries that has
+ * a reference to @fib with an action GK_FWD_GRANTOR.
+ */
+static void
+gk_flush_a_grantor_in_flow_table(struct gk_fib *fib, struct gk_instance *instance)
+{
+	uint32_t next = 0;
+	int32_t index;
+	const struct ip_flow *key;
+	void *data;
+
+	index = rte_hash_iterate(instance->ip_flow_hash_table,
+		(void *)&key, &data, &next);
+	while (index >= 0) {
+		struct flow_entry *fe = &instance->ip_flow_entry_table[index];
+		if (fe->grantor_fib == fib) {
+			int ret = rte_hash_del_key(
+				instance->ip_flow_hash_table,
+				&fe->flow);
+			if (likely(ret >= 0))
+				memset(fe, 0, sizeof(*fe));
+			else if (ret == -EINVAL) {
+				RTE_LOG(ERR, HASH,
+					"gk: The parameters are invalid at %s",
+					__func__);
+			} else if (ret == -ENOENT) {
+				RTE_LOG(ERR, HASH,
+					"gk: The flow key is not found at %s",
+					__func__);
+			}
```

Comment:
By looking at the ``rte_hash_del_key()`` at DPDK, it only returns two possible negative values. Also, this function doesn't set the ``errno`` when error happens.